libs/nostarve_queue/Cargo.toml

```diff
@@ -0,0 +1,14 @@
+[package]
+name = "nostarve_queue"
```

Comment:
Why? I don't get why we have the `utils` crate instead of many small crates.

Reply:
Many small compilation units is the C approach, Rust has crates that combine multiple modules into one compilation unit.

Having too many crates means more overhead like:

* you having to maintain a Cargo.toml with a list of dependencies for each component.
* having a bunch of directories with a `src/`  directory that contains `lib.rs` is harder to navigate than a bunch of .rs files: you have to press expand on these directories in your editor two times (VS Code doesn't have the "if there is a single file in the directory just show that one" expansion that github has which would reduce it by one).
* It also causes more lines in Cargo.lock and the cargo build output.
* cargo's dependency resolution algorithm is exponential in the worst case. We are probably not hitting these worst case paths here, but in general it's preferable to give it less complex problems.
* These modules are so small that the startup overhead for rustc is probably bigger compared to compiling them in one crate, even if they run in parallel.

I'm not saying that we should put the entire neon git repo into one big crate, there is a balance as crates also give parallelism in compilation. It also makes a lot of sense to split off modules that use some large dependencies that the other modules don't need. But this doesn't seem to be the case here. We can split up the utils crate, but if we do, we should split it into two or three pieces at most, not more. Right now the utils module contributes only 3.7 seconds to build time, so we won't gain much build time by splitting it up into 10 pieces instead of 2 or 3.

Reply:
This is the `cargo c --timings` report on my box, for the last piece which contains the utils module:

![Screenshot_20231017_172331](https://github.com/neondatabase/neon/assets/19521842/1b9d2840-c8cf-4732-b8ab-af27dc0c98e8)